File: generated-code/example/csharp/Model/OneOf.cs
namespace TransGenTest.Model
{
    public abstract class OneOf
    {
        public abstract void WriteTo(System.IO.BinaryWriter writer);
        public static OneOf ReadFrom(System.IO.BinaryReader reader)
        {
            switch (reader.ReadInt32())
            {
                case OptionOne.TAG:
                    return OptionOne.ReadFrom(reader);
                case OptionTwo.TAG:
                    return OptionTwo.ReadFrom(reader);
                default:
                    throw new System.Exception("Unexpected tag value");
            }
        }

        public class OptionOne : OneOf
        {
            public const int TAG = 0;
            public int[] VecI32 { get; set; }
            public long LongInt { get; set; }
            public OptionOne() {}
            public OptionOne(int[] vecI32, long longInt)
            {
                this.VecI32 = vecI32;
                this.LongInt = longInt;
            }
            public static new OptionOne ReadFrom(System.IO.BinaryReader reader)
            {
                var result = new OptionOne();
                result.VecI32 = new int[reader.ReadInt32()];
                for (int i = 0; i < result.VecI32.Length; i++)
                {
                    result.VecI32[i] = reader.ReadInt32();
                }
                result.LongInt = reader.ReadInt64();
                return result;
            }
            public override void WriteTo(System.IO.BinaryWriter writer)
            {
                writer.Write(TAG);
                writer.Write(VecI32.Length);
                foreach (var VecI32Element in VecI32)
                {
                    writer.Write(VecI32Element);
                }
                writer.Write(LongInt);
            }
        }

        public class OptionTwo : OneOf
        {
            public const int TAG = 1;
            public int Value { get; set; }
            public OptionTwo() {}
            public OptionTwo(int value)
            {
                this.Value = value;
            }
            public static new OptionTwo ReadFrom(System.IO.BinaryReader reader)
            {
                var result = new OptionTwo();
                result.Value = reader.ReadInt32();
                return result;
            }
            public override void WriteTo(System.IO.BinaryWriter writer)
            {
                writer.Write(TAG);
                writer.Write(Value);
            }
        }
    }
}

File: generated-code/python/example/TcpReadWrite/one_of.py
from stream_wrapper import StreamWrapper
from typing import List

class OneOf:
    """Oneof example"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "OneOf":
        """Read OneOf from input stream
        """
        tag = stream.read_int()
        if tag == OptionOne.TAG:
            return OneOf.OptionOne.read_from(stream)
        if tag == OptionTwo.TAG:
            return OneOf.OptionTwo.read_from(stream)
        raise Exception("Unexpected tag value")

class OptionOne(OneOf):
    """First option"""

    TAG = 0

    __slots__ = ("vec_int","long_int",)

    vec_int: List[int]
    long_int: int

    def __init__(self, vec_int: List[int], long_int: int):
        self.vec_int = vec_int
        """List of integers"""
        self.long_int = long_int
        """Long integer"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "OptionOne":
        """Read OptionOne from input stream
        """
        vec_int = []
        for _ in range(stream.read_int()):
            vec_int_element = stream.read_int()
            vec_int.append(vec_int_element)
        long_int = stream.read_long()
        return OptionOne(vec_int, long_int)
    
    def write_to(self, stream: StreamWrapper):
        """Write OptionOne to output stream
        """
        stream.write_int(self.TAG)
        stream.write_int(len(self.vec_int))
        for element in self.vec_int:
            stream.write_int(element)
        stream.write_long(self.long_int)
    
    def __repr__(self):
        return "OptionOne(" + \
            repr(self.vec_int) + \
            ", " + \
            repr(self.long_int) + \
            ")"

OneOf.OptionOne = OptionOne

class OptionTwo(OneOf):
    """Second option"""

    TAG = 1

    __slots__ = ("value",)

    value: int

    def __init__(self, value: int):
        self.value = value
        """usize"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "OptionTwo":
        """Read OptionTwo from input stream
        """
        value = stream.read_int()
        return OptionTwo(value)
    
    def write_to(self, stream: StreamWrapper):
        """Write OptionTwo to output stream
        """
        stream.write_int(self.TAG)
        stream.write_int(self.value)
    
    def __repr__(self):
        return "OptionTwo(" + \
            repr(self.value) + \
            ")"

OneOf.OptionTwo = OptionTwo